### ISSUE TITLE
WASM build files are corrupt after first build 

### ISSUE THREAD

**Issue by sno2** [ziglang/zig#11928]
### Zig Version

0.10.0-dev.2344+1f5b0c115

### Steps to Reproduce

Create a `build.zig` with the following content:

```zig
const std = @import("std");
pub fn build(b: *std.build.Builder) void {
    const mode = b.standardReleaseOptions();
    const lib = b.addSharedLibrary("main", "main.zig", b.version(0, 0, 0));
    lib.setBuildMode(mode);
    lib.setTarget(.{ .cpu_arch = .wasm32, .os_tag = .freestanding });
    lib.setOutputDir("zig-out");
    b.default_step.dependOn(&lib.step);
}
```

And have a `main.zig` file like this:

```zig
pub export fn add(a: i32, b: i32) i32 {
  return a + b;
}
```

Now, run `zig build` two times.

### Expected Behavior

The `zig-out` folder should contain the correct WASM files.

### Actual Behavior

The `zig-out` contains the correct WASM files after the first build, but after the second it contains symlinks with weird file names:

```
PS C:\projects\deno\cosmo\zig-out> ls


    Directory: C:\projects\deno\cosmo\zig-out


Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
-a---l         6/24/2022   4:56 PM              0 bZKGSWlMXpFbh1eQ
-a----         6/24/2022   4:52 PM         286106 cosmo.o
-a---l         6/24/2022   4:56 PM              0 kwnJm2-vC2ioqxj0
-a----         6/24/2022   4:52 PM         227137 libcosmo.so
```

**Comment by sno2** [comment]
Can confirm this still happens on `0.10.0-dev.2734+905a18849`.  The fix that I currently have is just delete the `zig-out` folder in my `build.zig` file.

**Comment by kubkon** [comment]
This is actually a bug in `std.os.renameatW` in that it follows symlinks by default which it shouldn't. I've pushed a fix in #12461 if you'd like to check it out @sno2. I've also verified locally that your repro no longer crashes after the first `zig build`.
